jars/src/org/toast/activegates/ActivateGate.kt

```diff
@@ -0,0 +1,38 @@
+package org.toast.activegates
+
+import com.fs.starfarer.api.campaign.InteractionDialogAPI
+import com.fs.starfarer.api.campaign.rules.MemoryAPI
+import com.fs.starfarer.api.impl.campaign.rulecmd.BaseCommandPlugin
+import com.fs.starfarer.api.util.Misc
+
+class ActivateGate : BaseCommandPlugin() {
+
+    override fun execute(ruleId: String, dialog: InteractionDialogAPI?,
+                         params: List<Misc.Token>,
+                         memoryMap: Map<String, MemoryAPI>): Boolean {
+
+        if (dialog == null) return false
+
+        val textPanel = dialog.textPanel
+
+        val gate = dialog.interactionTarget
+        if (gate.hasTag(GateCommandPlugin.TAG_GATE)) {
+            if (!gate.hasTag(GateCommandPlugin.TAG_GATE_ACTIVATED)) {
+                if (GateCommandPlugin.canActivate()) {
+                    GateCommandPlugin.payActivationCost()
+                    gate.addTag(GateCommandPlugin.TAG_GATE_ACTIVATED)
+                    gate.memory
+                    textPanel.addParagraph("Your crew offload the resources and, almost reverently, a single gamma core." +
```

Comment:
if the number of cores is changed, this text will also need to change

Reply:
Yep. I don't like it but want to prioritize UX over maintainability - I think the best solution might be to move all visible strings to a single class so it's less likely that one would get missed if something changes.